execution.go

```diff
@@ -0,0 +1,172 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package execution
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/dop251/goja"
+
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/lib"
+)
+
+func init() {
+	modules.Register("k6/x/execution", new(Execution))
```

Comment:
I considered naming it `k6/x/ecution` to avoid the stutter :smile:

Reply:
I do prefer if we followed the pattern from [xk6-segment](https://github.com/MStoykov/xk6-segment)

Where there was package with just the code for the extension and another one (in this case the "root" one) for the registering. 
This way it is possible to do what we currently do with the kafka output, where the code for the output is outside and we just import it in k6. 

This has 2 benefits IMO:
1. the cycle of the development for the extension isn't required to align with that for k6. This can mean that a more feature full version of the extension exist that will be then added to k6 core later on.
2. it lets people fork it and add their stuff way easier, possibly making a PR just to the extension repo.

I am not saying that we will develop like that this extension or even any future one, but given that it's just a 1 more package and 1 more file I find it's worth the additional work :).

Reply:
Good idea @MStoykov, see the latest push.

Importing it in k6 would also allow testing for it there. Should I move the tests back then? They really belong here, but if we can avoid having to tweak them and since this will eventually be part of core, we might as well leave them there.